### PR TITLE
create a new token on login and expire token

### PR DESCRIPTION
**Pull request by codenakama**
@warrenbuckley  can you please review these changes?

### CODE REVIEW

**Comment by biapar** [comment]
@codenakama I like that feature. How to add to my code?

**Comment by codenakama** [comment]
@biapar  I haven't had any response on this project so I decided to give another try to Matts Brailsford AuthU package, which @warrenbuckley  also recommended. I think you should go with this one and just make sure you are targeting .NET 4.6.5 (I think this is the current target version for this package) - https://github.com/mattbrailsford/umbraco-authu

It should give you all you need for JWT and token validation.

**Comment by biapar** [comment]
@codenakama UaU...I don't know of that new project! Is't possible to integrate to AD authentication?
Have you tried with Angular 4 client?

**Comment by codenakama** [comment]
@biapar  currently using for a new project at the company I work for. So far so good and it does all we required, and we using with React so yeah you could use it with angular or any js framework really. It doesnt work with AD I believe, it works with the umbraco authentication (users and members). But raise an issue on the project askign any questions you may have, Matt usually replies very swiftly.

**Comment by warrenbuckley** [comment]
Yeh sorry @codenakama I suck at getting time to review really old pet projects I created.
Let me see if I can find some time to look at this & merge this in.

File: Umbraco.AuthTokens/Data/UmbracoAuthTokenFactory.cs [hunk 1]
```diff
@@ -18,20 +18,26 @@ public static UmbracoAuthToken GenerateUserAuthToken(UmbracoAuthToken authToken)
         {
             //Date Time
             var dateCreated = DateTime.UtcNow;
+           
             var dateCreatedToString = dateCreated.ToString("u");
+            //this probably should be allowed to be configurable but keeping it simple and reasonable for now
+            var dateExpires = dateCreated.Add(TimeSpan.FromMinutes(60));
```
Comment (warrenbuckley) [inline Umbraco.AuthTokens/Data/UmbracoAuthTokenFactory.cs]: Yes for now this is fine @codenakama 
But I agree with your comment that it should be configurable in the future as you may not want to have a token with an hour expiry and may wish to refresh/get a new token more often

**Pull request merged**